### PR TITLE
API documentation for authorization profile endpoint

### PR DESCRIPTION
**Pull request by bdpiparva**
Todo

- [x] [Version check](https://github.com/bdpiparva/api.go.cd/blob/f0c368b29f956409d2a066ba1b14148ec25bea37/source/includes/authorization_profile/_00-describe.md.erb#L2) for all pages

- [x] Must have reference to api.gocd.io instead of api.go.cd

### CODE REVIEW

**Comment by jyotisingh** [comment]
@bdpiparva - rebase this please

**Comment by ketan** [comment]
@varshavaradarajan @bdpiparva — can we get this merged?

File: source/includes/authorization_configuration/_30-create.md.erb [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      "key": "ManagerDN",
+      "value": "uid=admin,ou=system"
+    },
+    {
+      "key": "SearchFilter",
+      "value": "uid"
+    },
+    {
+      "key": "Password",
+      "value": "secret"
```
Comment (varshavaradarajan) [inline source/includes/authorization_configuration/_30-create.md.erb]: encrypted_value

File: source/includes/authorization_configuration/_40-update.md.erb [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      "key": "ManagerDN",
+      "value": "uid=admin,ou=system"
+    },
+    {
+      "key": "SearchFilter",
+      "value": "uid"
+    },
+    {
+      "key": "Password",
+      "value": "secret"
```
Comment (varshavaradarajan) [inline source/includes/authorization_configuration/_40-update.md.erb]: encrypted_value

**Comment by jyotisingh** [comment]
Does it make sense to add a sample error response too for the benefit of the API consumer? Something like below. 
Response:
```
{
  "message": "Validations failed for authConfig 'ldap'. Error(s): [Validation failed.]. Please correct and resubmit.",
  "data": {
    "id": "ldap",
    "plugin_id": "cd.go.authentication.ldap",
    "properties": [
      {
        "key": "Url",
        "value": "ldap://ldap.server.url:389"
      },
      {
        "errors": {
          "SearchBases": [
            "SearchBases must not be blank."
          ]
        },
        "key": "SearchBases"
      },
      {
        "errors": {
          "UserLoginFilter": [
            "UserLoginFilter must not be blank."
          ]
        },
        "key": "UserLoginFilter"
      }
    ],
    "_links": {
      "self": {
        "href": "https://build.gocd.io/go/api/admin/security/auth_configs/ldap"
      },
      "doc": {
        "href": "https://api.gocd.io/#authorization-configuration"
      },
      "find": {
        "href": "https://build.gocd.io/go/api/admin/security/auth_configs/:auth_config_id"
      }
    }
  }
}
```

File: source/includes/authorization_configuration/_30-create.md.erb [hunk 3]
```diff
@@ -0,0 +1,87 @@
+## Create an authorization configuration
+
+```shell
+$ curl 'https://ci.example.com/go/api/admin/security/auth_configs' \
+      -u 'username:password' \
+      -H 'Accept: <%= data.apis.versions.authorization_configurations %>' \
+      -H 'Content-Type: application/json' \
+      -X POST -d '{
+        "id": "ldap",
+        "plugin_id": "cd.go.authorization.ldap",
```
Comment (jyotisingh) [inline source/includes/authorization_configuration/_30-create.md.erb]: change this to `cd.go.authentication.ldap` instead?

File: source/includes/authorization_configuration/_30-create.md.erb [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    },
+    "doc": {
+      "href": "https://api.gocd.io/#authorization-configuration"
+    },
+    "find": {
+      "href": "https://ci.example.com/go/api/admin/security/auth_configs/:auth_config_id"
+    }
+  },
+  "id": "ldap",
+  "plugin_id": "cd.go.authorization.ldap",
```
Comment (jyotisingh) [inline source/includes/authorization_configuration/_30-create.md.erb]: cd.go.authentication.ldap?

File: source/includes/_186-authorization-configurations.md.erb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# Authorization Configuration
+
+The Authorization configuration API allows admin users to manage authorization configuration, The authorization configuration will later be used by a plugin to authorize a user in the GoCD server.
```
Comment (jyotisingh) [inline source/includes/_186-authorization-configurations.md.erb]: That should be a full stop, not comma.

Change `, The authorization configuration will later...` to `. , This config will later...`

**Comment by bdpiparva** [comment]
> Does it make sense to add a sample error response too for the benefit of the API consumer? Something like below.

We are not documented validation failure use cases for other APIs.

**Pull request merged**